src/main/java/minicraft/screen/AchievementsDisplay.java

```diff
@@ -135,12 +135,15 @@ public void render(Screen screen) {
      * @return True if setting the achievement was successful.
      */
     public static boolean setAchievement(String id, boolean unlocked) {
-        return setAchievement(id, unlocked, true);
+        return setAchievement(id, unlocked, true, true);
     }
+	public static boolean setAchievement(boolean checkCreative, String id, boolean unlocked) { return setAchievement(id, unlocked, true, checkCreative); }
 
-    private static boolean setAchievement(String id, boolean unlocked, boolean save) {
+	private static boolean setAchievement(String id, boolean unlocked, boolean save, boolean checkCreative) {
         Achievement a = achievements.get(id);
 
+		// Return if it is in creative mode
+		if (checkCreative && !Game.isMode("creative")) return false;

```

Comment:
This will return if it should check and we are not in creative. Completely the opposite of what was the intended solution